From: Carlos Garnacho <carlosg@gnome.org>
Date: Mon, 7 Dec 2020 18:10:58 +0100
Subject: clutter: Drop CLUTTER_CLIENT_MESSAGE event

This is used nowhere and emitted nowhere. We can do without it.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1623>
---
 clutter/clutter/clutter-actor.c             | 6 ------
 clutter/clutter/clutter-enums.h             | 2 --
 clutter/clutter/clutter-event.c             | 5 -----
 clutter/clutter/clutter-main.c              | 3 ---
 src/core/events.c                           | 3 +--
 src/tests/clutter/interactive/test-events.c | 3 ---
 src/tests/clutter/interactive/test-grab.c   | 3 ---
 7 files changed, 1 insertion(+), 24 deletions(-)

diff --git a/clutter/clutter/clutter-actor.c b/clutter/clutter/clutter-actor.c
index febfb31..d4d1b5d 100644
--- a/clutter/clutter/clutter-actor.c
+++ b/clutter/clutter/clutter-actor.c
@@ -1099,7 +1099,6 @@ static GQuark quark_button = 0;
 static GQuark quark_scroll = 0;
 static GQuark quark_stage = 0;
 static GQuark quark_destroy = 0;
-static GQuark quark_client = 0;
 static GQuark quark_touch = 0;
 static GQuark quark_touchpad = 0;
 static GQuark quark_proximity = 0;
@@ -6026,7 +6025,6 @@ clutter_actor_class_init (ClutterActorClass *klass)
   quark_scroll = g_quark_from_static_string ("scroll");
   quark_stage = g_quark_from_static_string ("stage");
   quark_destroy = g_quark_from_static_string ("destroy");
-  quark_client = g_quark_from_static_string ("client");
   quark_touch = g_quark_from_static_string ("touch");
   quark_touchpad = g_quark_from_static_string ("touchpad");
   quark_proximity = g_quark_from_static_string ("proximity");
@@ -12715,10 +12713,6 @@ clutter_actor_event (ClutterActor       *actor,
       signal_num = -1;
       detail = quark_destroy;
       break;
-    case CLUTTER_CLIENT_MESSAGE:
-      signal_num = -1;
-      detail = quark_client;
-      break;
     case CLUTTER_STAGE_STATE:
       signal_num = -1;
       detail = quark_stage;
diff --git a/clutter/clutter/clutter-enums.h b/clutter/clutter/clutter-enums.h
index 9b0c29f..7a928d4 100644
--- a/clutter/clutter/clutter-enums.h
+++ b/clutter/clutter/clutter-enums.h
@@ -818,7 +818,6 @@ typedef enum /*< flags prefix=CLUTTER_EVENT >*/
  * @CLUTTER_SCROLL: Pointer scroll event
  * @CLUTTER_STAGE_STATE: Stage state change event
  * @CLUTTER_DESTROY_NOTIFY: Destroy notification event
- * @CLUTTER_CLIENT_MESSAGE: Client message event
  * @CLUTTER_TOUCH_BEGIN: A new touch event sequence has started;
  *   event added in 1.10
  * @CLUTTER_TOUCH_UPDATE: A touch event sequence has been updated;
@@ -855,7 +854,6 @@ typedef enum /*< prefix=CLUTTER >*/
   CLUTTER_SCROLL,
   CLUTTER_STAGE_STATE,
   CLUTTER_DESTROY_NOTIFY,
-  CLUTTER_CLIENT_MESSAGE,
   CLUTTER_TOUCH_BEGIN,
   CLUTTER_TOUCH_UPDATE,
   CLUTTER_TOUCH_END,
diff --git a/clutter/clutter/clutter-event.c b/clutter/clutter/clutter-event.c
index dba5d25..7af9f13 100644
--- a/clutter/clutter/clutter-event.c
+++ b/clutter/clutter/clutter-event.c
@@ -386,7 +386,6 @@ clutter_event_get_position (const ClutterEvent *event,
     case CLUTTER_KEY_RELEASE:
     case CLUTTER_STAGE_STATE:
     case CLUTTER_DESTROY_NOTIFY:
-    case CLUTTER_CLIENT_MESSAGE:
     case CLUTTER_EVENT_LAST:
     case CLUTTER_PROXIMITY_IN:
     case CLUTTER_PROXIMITY_OUT:
@@ -464,7 +463,6 @@ clutter_event_set_coords (ClutterEvent *event,
     case CLUTTER_KEY_RELEASE:
     case CLUTTER_STAGE_STATE:
     case CLUTTER_DESTROY_NOTIFY:
-    case CLUTTER_CLIENT_MESSAGE:
     case CLUTTER_EVENT_LAST:
     case CLUTTER_PROXIMITY_IN:
     case CLUTTER_PROXIMITY_OUT:
@@ -1058,7 +1056,6 @@ clutter_event_set_device (ClutterEvent       *event,
     case CLUTTER_NOTHING:
     case CLUTTER_STAGE_STATE:
     case CLUTTER_DESTROY_NOTIFY:
-    case CLUTTER_CLIENT_MESSAGE:
     case CLUTTER_EVENT_LAST:
     case CLUTTER_IM_COMMIT:
     case CLUTTER_IM_DELETE:
@@ -1158,7 +1155,6 @@ clutter_event_get_device (const ClutterEvent *event)
     case CLUTTER_NOTHING:
     case CLUTTER_STAGE_STATE:
     case CLUTTER_DESTROY_NOTIFY:
-    case CLUTTER_CLIENT_MESSAGE:
     case CLUTTER_IM_COMMIT:
     case CLUTTER_IM_DELETE:
     case CLUTTER_IM_PREEDIT:
@@ -1624,7 +1620,6 @@ clutter_event_get_axes (const ClutterEvent *event,
     case CLUTTER_NOTHING:
     case CLUTTER_STAGE_STATE:
     case CLUTTER_DESTROY_NOTIFY:
-    case CLUTTER_CLIENT_MESSAGE:
     case CLUTTER_ENTER:
     case CLUTTER_LEAVE:
     case CLUTTER_KEY_PRESS:
diff --git a/clutter/clutter/clutter-main.c b/clutter/clutter/clutter-main.c
index fd432dd..3dd5b0a 100644
--- a/clutter/clutter/clutter-main.c
+++ b/clutter/clutter/clutter-main.c
@@ -1917,9 +1917,6 @@ _clutter_process_event_details (ClutterActor        *stage,
           clutter_stage_event (CLUTTER_STAGE (stage), event);
         break;
 
-      case CLUTTER_CLIENT_MESSAGE:
-        break;
-
       case CLUTTER_DEVICE_ADDED:
       case CLUTTER_DEVICE_REMOVED:
         _clutter_event_process_filters (event);
diff --git a/src/core/events.c b/src/core/events.c
index e952220..16e6948 100644
--- a/src/core/events.c
+++ b/src/core/events.c
@@ -111,8 +111,7 @@ handle_idletime_for_event (const ClutterEvent *event)
       event->type == CLUTTER_ENTER ||
       event->type == CLUTTER_LEAVE ||
       event->type == CLUTTER_STAGE_STATE ||
-      event->type == CLUTTER_DESTROY_NOTIFY ||
-      event->type == CLUTTER_CLIENT_MESSAGE)
+      event->type == CLUTTER_DESTROY_NOTIFY)
     return;
 
   core_monitor = meta_idle_monitor_get_core ();
diff --git a/src/tests/clutter/interactive/test-events.c b/src/tests/clutter/interactive/test-events.c
index 5faf685..633d71d 100644
--- a/src/tests/clutter/interactive/test-events.c
+++ b/src/tests/clutter/interactive/test-events.c
@@ -353,9 +353,6 @@ input_cb (ClutterActor *actor,
     case CLUTTER_DESTROY_NOTIFY:
       g_print ("[%s] DESTROY NOTIFY", clutter_actor_get_name (source_actor));
       break;
-    case CLUTTER_CLIENT_MESSAGE:
-      g_print ("[%s] CLIENT MESSAGE", clutter_actor_get_name (source_actor));
-      break;
     case CLUTTER_TOUCHPAD_PINCH:
       g_print ("[%s] TOUCHPAD PINCH", clutter_actor_get_name (source_actor));
       break;
diff --git a/src/tests/clutter/interactive/test-grab.c b/src/tests/clutter/interactive/test-grab.c
index 595aac2..3b3a268 100644
--- a/src/tests/clutter/interactive/test-grab.c
+++ b/src/tests/clutter/interactive/test-grab.c
@@ -65,9 +65,6 @@ debug_event_cb (ClutterActor *actor,
     case CLUTTER_DESTROY_NOTIFY:
       printf("[%s] DESTROY NOTIFY", source);
       break;
-    case CLUTTER_CLIENT_MESSAGE:
-      printf("[%s] CLIENT MESSAGE\n", source);
-      break;
     case CLUTTER_TOUCH_BEGIN:
       g_print ("[%s] TOUCH BEGIN", source);
       break;
